FWCore/PluginManager/bin/refresh.cc

```diff
@@ -141,13 +141,13 @@ int main(int argc, char** argv) try {
       //if asked to look at whole directory, then we can also remove missing files
       removeMissingFiles = true;
 
-      boost::filesystem::directory_iterator file(directory);
-      boost::filesystem::directory_iterator end;
+      std::filesystem::directory_iterator file(directory);
+      std::filesystem::directory_iterator end;
 
       path cacheFile(directory);
       cacheFile /= standard::cachefileName();
 
-      std::time_t cacheLastChange(0);
+      std::filesystem::file_time_type cacheLastChange;
```

Comment:
what is default value of `cacheLastChange` here?

Reply:
ok [a] allowed me to generate the plugin cache so issue here is with default value of `cacheLastChange` 

[a]
```
@@ -147,9 +147,11 @@ int main(int argc, char** argv) try {
       path cacheFile(directory);
       cacheFile /= standard::cachefileName();
 
+      bool force_cache = true;
       std::filesystem::file_time_type cacheLastChange;
       if (exists(cacheFile)) {
         cacheLastChange = last_write_time(cacheFile);
+         force_cache = false;
       }
       for (; file != end; ++file) {
         path filename(*file);
@@ -164,7 +166,7 @@ int main(int argc, char** argv) try {
           continue;
         }
 
-        if (last_write_time(filename) > cacheLastChange) {
+        if (force_cache || last_write_time(filename) > cacheLastChange) {
           files.push_back(stringName);
         }
       }

```

Reply:
> what is default value of `cacheLastChange` here?

The default value should correspond the the epoch of the "trivial clock" the implementation uses
https://en.cppreference.com/w/cpp/filesystem/file_time_type
https://en.cppreference.com/w/cpp/chrono/time_point/time_point
(details being implementation-defined)

Reply:
https://github.com/cms-sw/cmssw/pull/30961 should fix this.

Reply:
thanks for the fix